### PR TITLE
Fixed issue with dedent utility function when building with python win10

### PR DESCRIPTION
**Pull request by bengaspar**
Fix for build issue in dedent utility function when building with python
https://github.com/caffe2/caffe2/issues/2151
It isn't entirely clear to me why we would need to pipe that through std in via the command line vs using the first arg.  Please review and let me know if this can cause any side effects.  Thank you for the review.

### CODE REVIEW

**Comment by pjh5** [comment]
Thanks for doing this. Can you rebase off of master to remove all the changes in caffe2/operators/conv_op_cudnn.cc  , which have been checked in in #2222 ?

We don't have CI set up yet to test this, but I'll validate that this works tomorrow morning when I'm back at my Windows desktop.

File: scripts/build_windows.bat [hunk 1]
```diff
@@ -8,14 +8,15 @@
 setlocal
 
 SET ORIGINAL_DIR=%cd%
-SET CAFFE2_ROOT=%~dp0%..
+SET CAFFE2_ROOT=%~dp0..
+SET PYBIND_DIR="%CAFFE2_ROOT%\third_party\pybind11\include"
```
Comment (pjh5) [inline scripts/build_windows.bat]: Is this needed for the dedent error is this an unrelated change?

**Comment by pjh5** [comment]
I think you'll have to add https://github.com/caffe2/caffe2.git as an upstream remote of your fork and then fetch that and rebase off of it's master in order to update this diff's base commit. I think it
`git remote add upstream https://github.com/caffe2/caffe2.git`
`git fetch upstream`
`git rebase upstream/master`

File: scripts/build_windows.bat (hunk 1 above)
Comment (bengaspar) [reply scripts/build_windows.bat]: No, this was just a pybind11 issue I was having, seemed to fix it but I'm not sure if its just my environment.

**Comment by bengaspar** [comment]
Sorry about the mess, first PR.  The diff should only be my change at this point.  Going forward I will be sure  to isolate the PR changes into a separate branch away from my other changes.  Thanks for helping me learn this process.

**Comment by orionr** [comment]
@caffe2bot test this please

**Comment by Erotemic** [comment]
@bengaspar as the original comment indicates, the use of stdin/stdout was to prevent needing to escape special characters if you were to place the text directly in the command. However, I think your solution also addresses those issues. So, it shouldn't cause any side effects. I simply didn't consider using `sys.argv`. I think your solution is cleaner better than my original code. Thanks for submitting the patch.  

I still would like to know what the actual issue was. I was never able to reproduce it on my windows box. Under what circumstances does windows have trouble piping stdout?

**Comment by yinghai** [comment]
@caffe2bot retest this please

**Comment by orionr** [comment]
Thank you for your contribution! PyTorch and Caffe2 are now officially merged with the PyTorch 1.0 prerelease. Please submit on https://github.com/pytorch/pytorch if this is still an issue over there.
